fuseimg/utils/visualization/imaging_multi_plot.py

```diff
@@ -0,0 +1,99 @@
+from typing import List , Callable , Any
+import matplotlib.pyplot as plt
+import numpy as np
+from functools import partial
+from fuseimg.utils.typing.typed_element import TypedElement
+from matplotlib.collections import PatchCollection
+from matplotlib.patches import Polygon
+
+def show_multiple_images(plot_seg : Callable , imgs : List, base_resolution :int = 5, **args ):
+    '''
+    Show multiple images with shared zoom/translation controls
+    everything passed as kwargs (for example - cmap='gray') will be passed to the individual imshow calls
+    special possible values are:
+        * cmap[image_index] = value
+            for example - cmap0='gray' will only change the cmap of the first image
+        * unify_size = 'dont_care'
+            will make sure that all images are resized to the same size
+    example usage:
+    imshowmultiple([img1, img2])
+    imshowmultiple(img1,img2,img3, cmap='gray', vmin=0.0, vmax=1.0)
+    imshowmultiple(img1,img2,img3, cmap0='gray')   #will use grayscale color map only on the first image and default cmap on the rest
+    imshowmultiple(img1,img2,img3, unify_size='blah')   #will resize all images to match
+    :param plot_seg : function to plot the ground truth segmentation
+    :param imgs: list of images in TypedElement format
+    :param base_resolution : base pixel resolution we want to maintain per image
+    :param args: additional parameters to the plot function of matplotlib
+    :return:
+    '''
+    assert len(imgs)>0
+    grid_size = int(np.sqrt(len(imgs))) + 1
+    fig = plt.figure(figsize=(base_resolution*grid_size,base_resolution*grid_size))
+    axis = []
+    
+    do_not_pass = ['unify_size','color']
+    do_not_pass += [ 'cmap'+str(i) for i in range(20)]
+    pass_kwargs = { k:d for k,d in args.items() if k not in do_not_pass }
+    for i,m in enumerate(imgs):
+
+        im = m.image
+        if 0==i:
+            axis.append(fig.add_subplot(grid_size,grid_size,i+1))
+        else:
+            axis.append(fig.add_subplot(grid_size, grid_size, i + 1, sharex=axis[0],sharey=axis[0]))
+            
+        img_cmap = 'cmap'+str(i+1)
+        if img_cmap in args:
+            pass_kwargs['cmap'] = args[img_cmap]      
+            axis[i].imshow(im, interpolation='none', **pass_kwargs)
+        else:
```

Comment:
else you use the cmap from the last iteration?

Reply:
yes